sopel/bot.py

```diff
@@ -702,6 +703,15 @@ def call_rule(
 
         try:
             rule.execute(sopel, trigger)
+        except PluginAbort as abort:
+            LOGGER.warning("Plugin aborted while handling trigger %r", trigger, exc_info=abort)
+
+            if len(abort.args) == 1:
```

Comment:
I would go as far as formalizing the constructor signature & attributes of `PluginAbort()`:

```py
def __init__(self, message: str | None = None):
    self.message = message
    super().__init__(message)
```

As mentioned in other notes, the purpose of this feature is pretty specific. I don't understand why it needs to take arbitrary positional + keyword args.